common.go

```diff
@@ -434,3 +441,16 @@ func GetLanguageByAlias(alias string) (lang string, ok bool) {
 
 	return
 }
+
+// GetLanguagesByGitattributes returns a length 1 slice with the language assigned in .gitattributes if the regular expresion
+// matchs with the filename. It is comply with the signature to be a Strategy type.
+func GetLanguagesByGitattributes(filename string, content []byte, candidates []string) []string {
+	languages := []string{}
```

Comment:
Maybe it is better, to clarify the interface to remove this allocation from here and move it to the return statement, as otherwise we are allocating a slice to return it empty in case the filename does not match any languageGitattributes.